nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -80,6 +84,13 @@ public static DocumentModel moveContentToColdStorage(CoreSession session, Docume
         DocumentModel documentModel = session.getDocument(documentRef);
         log.debug("Move to cold storage the main content of document: {}", documentModel);
 
+        if (!session.hasPermission(documentRef, SecurityConstants.WRITE_COLD_STORAGE)) {
+            log.debug("The user {} does not have the right permissions to move the content of document {}",
+                    session.getPrincipal(), documentModel);
```

Comment:
Can you please use a lambda here to avoid systematically fetching the principal?
```
session::getPrincipal, () -> documentModel
```